### ISSUE TITLE
format: checkbox does not send info over POST request but selection2 does

### ISSUE THREAD

**Issue by PydPiper** [json-editor/json-editor#1103]
I don't know if this is a bug or i am missing something but when i format a field to `format: selection2` and send my form in through a `POST` request. I get the selections information in the request, but when i switch the `format: checkbox` the schema renders fine and works but the `POST` doesnt send its selections through the request post. Any thoughts as to what could be causing this?

**Comment by schmunk42** [comment]
Can you provide an example schema?

**Comment by PydPiper** [comment]
Here is a sample django project setup to demo this:

in urls.py

```python
urlpatterns = [
    path('testing', views.testing, name='testing'),
]
```

in views.py

```python
def testing(request):
    print(request.POST)
    return render(request, 'test.html', {})
```

in html

```python
<head>
    <!-- Required import to render HTML without refreshing page using jquery -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
    <!-- Required import to use JSON Schemas -->
    <script src="https://cdn.jsdelivr.net/npm/@json-editor/json-editor@latest/dist/jsoneditor.min.js"></script>

</head>

<body>
    <form method="post">
        {% csrf_token %}
        <input type="submit" value="SUBMIT">
        <div id="placeholder"></div>
    </form>
</body>

<script>
    var Eproject = new JSONEditor(document.getElementById('placeholder'),{schema: {"type": "array","uniqueItems": true, "format": "checkbox",  "items": {    "type": "integer",    "enum": [      10,      100    ]  }}});
</script>
```

here is the response
```
<QueryDict: {}>
[27/Jan/2022 07:22:19] "GET /testing HTTP/1.1" 200 854
<QueryDict: {'csrfmiddlewaretoken': ['boEnSWyBcWEW1DQczf2H1gjYME7PhekB8ErWiRcaj1fEl80bRvEezuqxYTAjlU7e']}>
[27/Jan/2022 07:22:22] "POST /testing HTTP/1.1" 200 854
```

here is interactive copy of the schema, but you need a full project to see that the checkbox does not make it through a POST request

https://json-editor.github.io/json-editor/?data=N4Ig9gDgLglmB2BnEAuUMDGCA2MBGqIAZglAIYDuApomALZUCsIANOHgFZUZQD62ZAJ5gArlELwwAJzplsrEIgwALKrNSgogiFUJkpUoQpHwYARxFUAklDXIUUKZbYkZZcShAruAazxgADwUYWzp7TW1dTxh4WwBzKikFKngROlQAbQBGAAYWXJyAXQBfYrZEZTAKXkSpaXsQGNtDHjh4BShVBkJ/MChERzIIABYQUqA

**Comment by PydPiper** [comment]
I have also double checked that this was not a Django related issue since i observed the same thing when i created a dummy Flask project

**Comment by PydPiper** [comment]
For reference here is what a checkbox renders in html

![image](https://user-images.githubusercontent.com/35475223/151580450-fcc32896-f306-4838-9ddd-9a552416f298.png)

versus the checkbox

![image](https://user-images.githubusercontent.com/35475223/151580568-11847a85-8daa-4ff0-b946-e7e495dca2fc.png)

**Comment by PydPiper** [comment]
@schmunk42 any progress on this?

**Comment by germanbisurgi** [comment]
I don't see anywhere in your code the [select2](https://select2.org/) dependency

**Comment by PydPiper** [comment]
@germanbisurgi please reread the question. The problem is about checkbox.

**Comment by germanbisurgi** [comment]
do you send a stringified version if the editor value?

**Comment by schmunk42** [comment]
@PydPiper So you rely on the "native" form input values, rather than the "compiled" JSON from the editor?

From the source, it looks like editor checkbox `<input>` tags lack a `name="prop"` attribute.

**Comment by PydPiper** [comment]
@schmunk42 sorry i am not familiar with what you mean by compiled json. Is there a better way to send a post request with json schemas. The simplified example is how i am currently using the form (just on a much larger schema)

**Comment by germanbisurgi** [comment]
You kcan have a hidden input and listen to the change event of the editor. In the change callback you can set the value of the editor in that input and only send that field data per POST.

```javascript
editor.on('change', function () {
  document.getElementById('my-input').value = JSON.stringify(editor.getValue());
})
```

**Comment by PydPiper** [comment]
thanks @germanbisurgi this seems to be a work around. I think it is a good idea to keep this issue open until the checkbox POST is fixed

**Comment by germanbisurgi** [comment]
Yes, i will put a bug label too

**Comment by schmunk42** [comment]
> @schmunk42 sorry i am not familiar with what you mean by compiled json. Is there a better way to send a post request with json schemas. The simplified example is how i am currently using the form (just on a much larger schema)

We're just using the editor JSON value as shown in the textarea on the right side in the [example](https://json-editor.github.io/json-editor/).

Why would you need the single inputs in a post request?

**Comment by PydPiper** [comment]
@schmunk42 the same way as you can have active selections in a select2 format, the checkbox should behave the same with it’s options that are checked

**Comment by germanbisurgi** [comment]
Yeah, actually that is the intended use for the json-editor. It is not actually a bug. If you think about it. In html there are no inputs for array or objects. Another common use for it is to store stringified json data in localstorage.

**Comment by PydPiper** [comment]
then i don’t understand the use for checkbox format if you cannot access what was selected on the backend

**Comment by PydPiper** [comment]
I am surprised this is not a bug, but i’ll close out this issue if the intended function is not to be able to access what was check on the backend. 

I would suggest to put this issue as a warning in the docs about format=checkbox

**Comment by germanbisurgi** [comment]
I think we should expand the introduction in the readme to better clarify what this library does and show some common implementations as well.
